lib/internal/modules/esm/module_job.js

```diff
@@ -151,7 +167,30 @@ class ModuleJob {
     await this.instantiate();
     const timeout = -1;
     const breakOnSigint = false;
-    await this.module.evaluate(timeout, breakOnSigint);
+    try {
+      await this.module.evaluate(timeout, breakOnSigint);
+    } catch (e) {
+      if (e.name === 'ReferenceError' &&
+          StringPrototypeEndsWith(e.message, ' is not defined') &&
+          isCommonJSGlobalLikeNotDefinedError(e.message)) {
+        e.message += ' in ES module scope';
+
+        if (StringPrototypeStartsWith(e.message, 'require')) {
+          e.message += '. You can use `await import` instead';
+        }
+
+        if (StringPrototypeEndsWith(this.module.url, '.js') &&
+            StringPrototypeStartsWith(this.module.url, 'file://')) {
+          e.message +=
+            '. It seems you are trying to load a file using `.js` extension ' +
+            'inside a folder containing a `package.json` that includes ' +
+            '`"type": "module"`; you need to use the `.cjs` extension to ' +
+            'load the file as a CommonJS module, or add a `package.json` ' +
```

Comment:
if there's already a package.json with type module, then "add a package.json" is a bit confusing :-/

Reply:
Hum indeed. I'm assuming people in this situation have a `package.json` in a parent folder which is setting `"type": "module"`, so in this case the right thing to do is to add another `package.json` in the same folder as the `.js` file. Do you have a suggestion how to put this in comprehensible English sentence?

Reply:
since this is on the error path can we just get the package.json scope we are in and show that?

Reply:
> since this is on the error path can we just get the package.json scope we are in and show that?

Done in 9001e34.

Reply:
@aduh95 another "right thing to do" is to remove type:module so that `.js` files mean CJS.

Reply:
@ljharb the reason I didn't add this suggestion is removing the `"type"` in a parent folder `package.json` is likely to break things – assuming it was added for a reason. Also Gus is already complaining the message is too long, do you think it's necessary to add that other workaround?

Reply:
If the user is encountering this error, then I would bet money that this is one of the first files they've created in the repo - they probably followed a tutorial that blindly told them to use type module. I take your point, though.